server/service/queries.go

```diff
@@ -459,15 +461,11 @@ func applyQuerySpecsEndpoint(ctx context.Context, request interface{}, svc fleet
 }
 
 func (svc *Service) ApplyQuerySpecs(ctx context.Context, specs []*fleet.QuerySpec) error {
+	// check that the user can create queries
 	if err := svc.authz.Authorize(ctx, &fleet.Query{}, fleet.ActionWrite); err != nil {
 		return err
 	}
 
-	vc, ok := viewer.FromContext(ctx)
```

Comment:
That's removed because it can't happen after an `Authorize` call, is that it? Not sure why that was there...

Reply:
I moved it a little bit further down where `vc` is first used. I thought this would read better ie co-locate code.

Reply:
Oh I see! Hmm in this case I think it'd be better to keep it up here, to prevent having to load and verify all queries in the specs if there is no authenticated user.